M2/Macaulay2/m2/methods.m2

```diff
@@ -1,159 +1,262 @@
+
 --		Copyright 1993-2002 by Daniel R. Grayson
+-*
+  Methods are the most common type of dynamic dispatch, along with Hooks.
+  The method function is stored under the youngest mutable hash table in the
+  method key sequence.
+*-
 
 needs "option.m2"
 
-MethodFunction = new Type of CompiledFunctionClosure
-MethodFunctionSingle = new Type of CompiledFunctionClosure
-MethodFunctionBinary = new Type of CompiledFunctionClosure
-MethodFunctionWithOptions = new Type of FunctionClosure
+-----------------------------------------------------------------------------
+-- Local variables
+-----------------------------------------------------------------------------
 
-MethodFunction.synonym = "method function"
-MethodFunctionSingle.synonym = "method function with a single argument"
-MethodFunctionBinary.synonym = "binary method function"
-MethodFunctionWithOptions.synonym = "method function with options"
+-----------------------------------------------------------------------------
+-- Local utilities
+-----------------------------------------------------------------------------
 
-dispatcherFunctions = {}
+-- see lists.m2
+all' := (L, p) -> not any(L, x -> not p x)
```

Comment:
Why not just use `all` here?

Reply:
As with `foldL`, it's because `all` is a method that isn't yet defined, and we can't use methods for defining methods.

Reply:
As with `fold`, if the method isn't called before "fold" is defined, it will work.

Reply:
What? `all'` is used in the function `method`
https://github.com/Macaulay2/M2/blob/8950020fd3018ed094e38cedb8738be5fede37a4/M2/Macaulay2/m2/methods.m2#L208-L211
and `all` is defined by calling the function `method`:
https://github.com/Macaulay2/M2/blob/8950020fd3018ed094e38cedb8738be5fede37a4/M2/Macaulay2/m2/lists.m2#L101
The literal definition of the method `all` will call the function `method` which needs `all'`. How will this work?

This conversation is hypocritical because you wrote `all'`:
https://github.com/Macaulay2/M2/blob/80fdea3b784101577cfea93d428043805520a2e6/M2/Macaulay2/m2/methods.m2#L130
I just replaced it with a more efficient version than yours!

Reply:
I'll resolve this conversation as I don't see a way to change anything here. You can of course undo it if you have another suggestion.